s3bucket/main.tf

```diff
@@ -0,0 +1,57 @@
+resource "aws_s3_bucket" "codedeploy_bucket" {
+  bucket = "${var.name_prefix}-codedeploy-releases"
+  acl    = "private"
+
+  versioning {
+    enabled = true
+  }
+}
+
+resource "aws_iam_policy" "codedeploy_policy" {
```

Comment:
But this doesn't create the instance profile, it only creates the policy. Then it's up to the caller of this module to attach that policy to the corresponding instance profile. In this case, it's done here: https://github.com/skyscrapers/viafutura/pull/74/files#diff-28fe565a8ae21d396549c0b739878395R149